lib/src/dap_resolver.dart

```diff
@@ -9,11 +13,152 @@ class DapResolver {
   DapResolver(this.didResolver, {http.Client? client})
       : client = client ?? http.Client();
 
-  Future<void> getMoneyAddresses(Dap dap) async {}
+  Future<List<MoneyAddress>> getMoneyAddresses(Dap dap) async {
+    throw UnimplementedError();
+  }
+
+  Future<DapResolutionResult> resolve(Dap dap) async {
+    final registryDidResolution = await didResolver.resolveDid(dap.registryDid);
+    if (registryDidResolution.hasError()) {
+      throw DapResolutionException(
+          'Failed to resolve registry DID: ${registryDidResolution.didResolutionMetadata.error}');
+    }
+
+    final registryServices = registryDidResolution.didDocument!.service ?? [];
+    final dapRegistryService = registryServices.firstWhere(
+      (service) => service.type == 'DAPRegistry',
+      orElse: () => throw DapResolutionException(
+          'Registry DID does not have a DAPRegistry service'),
+    );
```

Comment:
nit:
Since you're handing the `orElse` in the 2nd case, you might want to avoid force unwrapping the `didDocument!` in the first. Maybe something like:
```dart
final dapRegistryService = registryDidResolution.didDocument?.service.firstWhereOrNull((s) => s.type == 'DAPRegistry');
if (dapRegistryService == null) {
  throw DapResolutionException(
          'Registry DID does not have a DAPRegistry service');
}
```
(this code is probably wrong since I just wrote it in Github 😂 )

Reply:
There's a few other places where force unwarpping is happening here too. Either way is cool, but I tend to lean towards never force unwrapping expect in specific cases in widget layout trees